src/main/java/org/example/bookshop/service/impl/ShoppingCartManagerServiceImpl.java

```diff
@@ -0,0 +1,81 @@
+package org.example.bookshop.service.impl;
+
+import jakarta.persistence.EntityNotFoundException;
+import java.util.HashSet;
+import lombok.RequiredArgsConstructor;
+import org.example.bookshop.dto.cart.ShoppingCartDto;
+import org.example.bookshop.dto.cart.item.CreateCartItemRequestDto;
+import org.example.bookshop.dto.cart.item.UpdateCartItemRequestDto;
+import org.example.bookshop.exception.DuplicateException;
+import org.example.bookshop.mapper.CartItemMapper;
+import org.example.bookshop.mapper.ShoppingCartMapper;
+import org.example.bookshop.model.CartItem;
+import org.example.bookshop.model.ShoppingCart;
+import org.example.bookshop.repository.BookRepository;
+import org.example.bookshop.repository.CartItemRepository;
+import org.example.bookshop.service.ShoppingCartManagerService;
+import org.example.bookshop.service.ShoppingCartService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartManagerServiceImpl implements ShoppingCartManagerService {
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemRepository cartItemRepository;
+    private final CartItemMapper cartItemMapper;
+    private final ShoppingCartService shoppingCartService;
+    private final BookRepository bookRepository;
+
+    @Override
+    public ShoppingCartDto saveCartItem(
+            Pageable pageable, CreateCartItemRequestDto cartItemRequestDto) {
+        if (!bookRepository.existsById(cartItemRequestDto.getBookId())) {
```

Comment:
see the example of implementation
```suggestion
        Book book = bookRepository.findById(itemDto.bookId())
                .orElseThrow(() -> new EntityNotFoundException(
                        String.format("Book with id: %d is not found", itemDto.bookId())
                ));
        ShoppingCart cart = repository.findByUserId(userId);
        cart.getCartItems().stream()
                .filter(item -> item.getBook().getId().equals(itemDto.bookId()))
                .findFirst()
                .ifPresentOrElse(item -> item.setQuantity(item.getQuantity() + itemDto.quantity()),
                        () -> addCartItemToCart(itemDto, book, cart));
        repository.save(cart);
        return cartMapper.toDto(cart);
```